updated confusing virtual IP and service proxy (#46873)
kubernetes/website: content/en/docs/reference/networking/virtual-ips.md

addresses which are transparently redirected as needed.

When clients connect to the VIP, their traffic is automatically transported to an

> appropriate endpoint. The kubelet running on that node set up environment variable 

Reviewer: ```suggestion
appropriate endpoint. The `kubelet` running on that node sets up environment variables 
```

Author: accepted and updated as per comments

> for each active service based on Service's virtual IP address and port. Similarly,

Reviewer: ```suggestion
for each active Service based on the Service's virtual IP address and port. Similarly,
```

Author: accepted and updated as per comments

> DNS creates set of DNS records for each service by watching kubernetes API.

Reviewer: ```suggestion
DNS creates set of DNS records for each Service by watching the Kubernetes API.
```

Author: accepted and updated as per comments

> Either of two ways can be responsible for discovery of service.

Reviewer 2: SIG Docs here, the initial statement about the environmental variable is correct. (See [Services](https://kubernetes.io/docs/concepts/services-networking/service/#discovering-services)).

The second statement about DNS is also correct.

> Either of two ways can be responsible for discovery of service.

Reviewer 2: To try it out:
**service.yaml**
```
apiVersion: v1
kind: Service
metadata:
  name: myapp-service
spec:
  selector:
    app: myapp
  ports:
  - port: 80
    targetPort: 80
```
```
kubectl apply -f service.yaml
```
**deployment.yaml**
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: myapp
spec:
  replicas: 1
  selector:
    matchLabels:
      app: myapp
  template:
    metadata:
      labels:
        app: myapp
    spec:
      containers:
      - name: myapp
        image: nginx:latest
        ports:
        - containerPort: 80
```
```
kubectl apply -f deployment
```
**View Service**
```
kubectl get svc

NAME            TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)   AGE
kubernetes      ClusterIP   10.96.0.1        <none>        443/TCP   7d23h
myapp-service   ClusterIP   10.101.104.130   <none>        80/TCP    15m
```

**Exec into the pod and do an nslookup of the service dns name**
```
root@myapp-7799f86b6-4dcck:/# nslookup myapp-service.default.svc.cluster.local

;; Got recursion not available from 10.96.0.10
;; Got recursion not available from 10.96.0.10
;; Got recursion not available from 10.96.0.10
;; Got recursion not available from 10.96.0.10
Server:         10.96.0.10
Address:        10.96.0.10#53

Name:   myapp-service.default.svc.cluster.local
Address: 10.101.104.130
;; Got recursion not available from 10.96.0.10
```

**View the created environmental variable via an exec into the pod**
```
root@myapp-7799f86b6-4dcck:/# | grep -i MYAPP

HOSTNAME=myapp-7799f86b6-4dcck
MYAPP_SERVICE_PORT_80_TCP=tcp://10.101.104.130:80
MYAPP_SERVICE_PORT_80_TCP_PROTO=tcp
MYAPP_SERVICE_PORT_80_TCP_PORT=80
MYAPP_SERVICE_PORT=tcp://10.101.104.130:80
MYAPP_SERVICE_SERVICE_HOST=10.101.104.130
MYAPP_SERVICE_SERVICE_PORT=80
MYAPP_SERVICE_PORT_80_TCP_ADDR=10.101.104.130
```
@kundan2707, next time, it'd be great to support your claim by either showing a sample we can try, referencing an existing URL in the documentation, or a merged PR.
